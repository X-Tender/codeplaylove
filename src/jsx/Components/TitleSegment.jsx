import React from 'react';
import PropTypes from 'prop-types';
import MaskedSVG from './MaskedSVG';

const propTypes = {
	color: PropTypes.string.isRequired,
	children: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
};

const TitleSegment = ({ color, children, type }) => (
	<span className={`title__text title__text--${type}`}>
		<MaskedSVG color={color} text={children} type={type} />
	</span>
);

TitleSegment.propTypes = propTypes;

export default TitleSegment;
