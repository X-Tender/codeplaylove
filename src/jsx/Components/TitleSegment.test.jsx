import React from 'react';
import { shallow } from 'enzyme';
import TitleSegment from './TitleSegment';

describe('TitleSegment', () => {
	const color = '#ff000';
	const children = 'CODE';
	const type = 'code';

	const comp = shallow(
		<TitleSegment color={color} type={type}>
			{children}
		</TitleSegment>
	);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});

	it('renders props correctly', () => {
		expect(comp.find(`span.title__text--${type}`).exists()).toBe(true);
		expect(comp.find('MaskedSVG').prop('color')).toEqual(color);
		expect(comp.find('MaskedSVG').prop('type')).toEqual(type);
	});
});
