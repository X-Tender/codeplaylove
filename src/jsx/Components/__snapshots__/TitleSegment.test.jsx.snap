// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSegment renders correctly 1`] = `
<span
  className="title__text title__text--code"
>
  <MaskedSVG
    color="#ff000"
    text="CODE"
    type="code"
  />
</span>
`;
